internal/testutils/endpoint_test.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testutils
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/kubernetes-incubator/external-dns/endpoint"
+)
+
+func ExampleSameEndpoints() {
+	eps := []*endpoint.Endpoint{
+		{
+			DNSName: "example.org",
+			Target:  "load-balancer.org",
+		},
+		{
+			DNSName:    "example.org",
+			Target:     "load-balancer.org",
+			RecordType: "TXT",
+		},
+		{
+			DNSName:    "abc.com",
+			Target:     "something",
+			RecordType: "TXT",
+		},
+		{
+			DNSName:    "abc.com",
+			Target:     "1.2.3.4",
+			RecordType: "CNAME",
```

Comment:
yes, but endpoint itself has no knowledge regarding that, maybe makes sense to change that for consistency sake, but where it is relevant I was making sure to be correct with the recordtype